### ISSUE TITLE
Implement OneCycle Learning Rate Schedule

### ISSUE THREAD

**Issue by n2cholas** [deepmind/optax#19]
OneCycle learning rate from this paper: https://arxiv.org/abs/1708.07120. 

The paper describes a 3-stage piecewise linear schedule. [PyTorch](https://pytorch.org/docs/stable/optim.html#torch.optim.lr_scheduler.OneCycleLR) and [FastAI](https://fastai1.fast.ai/callbacks.one_cycle.html) implement a 2 stage variant with cosine annealing (which apparently works better in practice).

I would like to contribute both to this library, if you think they are suitable additions.

**Comment by mtthss** [comment]
This would be a beautiful contribution thanks!
